datafusion/expr/src/expr.rs

```diff
@@ -1623,6 +1623,24 @@ impl Expr {
                 | Expr::SimilarTo(Like { expr, pattern, .. }) => {
                     rewrite_placeholder(pattern.as_mut(), expr.as_ref(), schema)?;
                 }
+                Expr::InSubquery(InSubquery {
+                    expr,
+                    subquery,
+                    negated: _,
+                }) => {
+                    let subquery_schema = subquery.subquery.schema();
+                    // Using the datatype of the first field in the subquery
+                    if let Some(first_field) = subquery_schema.fields().first() {
```

Comment:
Nit: we might better want to report an error if subquery_schema contains multiple fields (unexpected), it might be problematic to troubleshot such case in the future if we just assume the first field

Reply:
Just writing what we talked about...

This function gets ran for every query, even if there are no placeholders. We don't want to error if we cannot infer for some case, we should just not infer.